### DISCUSSION TITLE
Waking up from deepsleep

### DISCUSSION THREAD

**Discussion by CrabbyPete** [micropython/micropython#13698] [RP2040 / Pico] [answered]
The ESP32 is able to wake from a deepsleep call on a pin interupt. Is it possible to do the same thing with deepsleep on a Pico?

**Comment by GitHubsSilverBullet** [comment, marked as answer]
The deepsleep wake-up works just fine on the PiPico.
Causes a Reset and the reset_cause can then be queried (it's an integer 3)

**Comment by bidrohini** [comment]
It's possible to wake a Raspberry Pi Pico from deep sleep using a pin interrupt, although it's not as straightforward as with the ESP32. [This video](https://www.youtube.com/watch?v=05o5QOZkpwg) may help you. For your convenience, you can make a [Raspberry Pi Pico breakout board](https://www.pcbway.com/project/shareproject/Raspberry_Pi_Pico_Breakout_Board__GPIO_extension_.html).

**Reply by GitHubsSilverBullet** [reply]
A video? Really? For a wakeup? One line of code?
The world is coming to an end…
